### DISCUSSION TITLE
Serving llama.cpp as an API

### DISCUSSION THREAD

**Discussion by evgenyfedorchenko** [ggml-org/llama.cpp#6501] [Q&A] [answered]
Dear Community,
Is there any possibility to serve llama.cpp on some port not like HTML form, but rather to allow sending JSON request with prompt/history and receiving the response in JSON format? (similar to ChatGPT API, for example).
Of course, i can write it manually for myself, but i wonder if there is any existing solution to not reinvent the wheel.
Appreciate your advises.

**Comment by phymbert** [comment, marked as answer]
Please have a look to the [server](https://github.com/ggerganov/llama.cpp/blob/master/tools/server/README.md)

**Reply by KissPeter** [reply]
404

**Reply by CISC** [reply]
Refresh. :)

**Comment by SunYingjie0228** [comment]
llama could cmake . to a server .but i sufer some questions,after cmake ,don't hava a server.txt ,this is a big queation to me.
